src/Digdir.Domain.Dialogporten.Infrastructure/InfrastructureExtensions.cs

```diff
@@ -56,24 +65,35 @@ public static class InfrastructureExtensions
             .AddValidatorsFromAssembly(thisAssembly, ServiceLifetime.Transient, includeInternalTypes: true);
 
         var infrastructureSettings = infrastructureConfigurationSection.Get<InfrastructureSettings>()
-        ?? throw new InvalidOperationException("Failed to get Redis settings. Infrastructure settings must not be null.");
+                    ?? throw new InvalidOperationException("Failed to get Redis settings. Infrastructure settings must not be null.");
+
+        services.AddFusionCacheNewtonsoftJsonSerializer();
 
         if (infrastructureSettings.Redis.Enabled == true)
         {
-            services.AddStackExchangeRedisCache(options =>
-            {
-                var infrastructureSettings = infrastructureConfigurationSection.Get<InfrastructureSettings>()
-                    ?? throw new InvalidOperationException("Failed to get Redis connection string. Infrastructure settings must not be null.");
-                var connectionString = infrastructureSettings.Redis.ConnectionString;
-                options.Configuration = connectionString;
-                options.InstanceName = "Redis";
-            });
+            services.AddStackExchangeRedisCache(opt => opt.Configuration = infrastructureSettings.Redis.ConnectionString);
+            services.AddFusionCacheStackExchangeRedisBackplane(opt => opt.Configuration = infrastructureSettings.Redis.ConnectionString);
         }
         else
         {
             services.AddDistributedMemoryCache();
         }
 
+        // todo: do we need default cache? 🤔
+        ConfigureFusionCache(services);
+        ConfigureFusionCache(services, nameof(Altinn.NameRegistry), new()
+        {
+            Duration = TimeSpan.FromDays(1),
+        });
+        ConfigureFusionCache(services, nameof(Altinn.ResourceRegistry), new()
+        {
+            Duration = TimeSpan.FromDays(1),
```

Comment:
Denne burde vi ta ned en hel del. Blir annoying for de som vil teste en ny-opprettet serviceResource å måtte vente ett døgn før de kan referere den i Dialogporten. Ifm #40 blir det endringer her, men inntil videre kan du redusere denne til 20 minutter.

Reply:
Yes!